### ISSUE TITLE
[bug]: No audio in call between 2 pjsip endpoints using opus codec

### ISSUE THREAD

**Issue by Dhanapathy** [asterisk/asterisk#950]
### Severity

Minor

### Versions

18.23, 20.9

### Components/Modules

chan_pjsip

### Operating Environment

OS is Ubuntu 22.04 LTS

### Frequency of Occurrence

Constant

### Issue Description

call between chan_pjsip endpoints using direct media and codec as opus, has no audio. But when I change codec to ulaw it works fine and also when I change chan_pjsip to chan_sip, direct media using opus works fine. When I use chan_pjsip and opus codec,  directmedia is not working.
[ast_verbose_full_PJSIPendpoints_Noaudio.txt](https://github.com/user-attachments/files/17410869/ast_verbose_full_PJSIPendpoints_Noaudio.txt)
[ast_verbose_full_SIPendpoints_workingFine.txt](https://github.com/user-attachments/files/17410870/ast_verbose_full_SIPendpoints_workingFine.txt)

### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

**Comment by gtjoseph** [comment]
Yeah, the dynamic payload types aren't being coordinated when the reINVITEs are being sent for direct media.

**Comment by m3-117** [comment]
On some endpoints, the payload type can be set to 107 as in asterisk, in which case directmedia is possible.

**Comment by alexbakker** [comment]
Just ran into this as well. I imagine this'll be challenging to support. For example, if the A leg selected payload type 107 for ``telephone-event``, and the B leg selected 107 for ``opus``, Asterisk would have to re-INVITE both legs to move them over to entirely different dynamic payload types for both codecs.

Related pull request (only tackles DTMF): https://github.com/asterisk/asterisk/pull/2017
